nifi-nar-bundles/nifi-aws-bundle/nifi-aws-abstract-processors/src/main/java/org/apache/nifi/processors/aws/wag/AbstractAWSGatewayApiProcessor.java

```diff
@@ -467,23 +469,22 @@ protected InputStream getRequestBodyToSend(final ProcessSession session,
 
     protected GenericApiGatewayRequestBuilder setHeaderProperties(final ProcessContext context,
                                                                   GenericApiGatewayRequestBuilder requestBuilder,
-                                                                  HttpMethodName methodName,
-                                                                  final FlowFile requestFlowFile) {
+                                                                  final HttpMethodName methodName,
+                                                                  final Map<String, String> requestAttributes) {
 
-        Map<String, String> headers = new HashMap<>();
-        for (String headerKey : dynamicPropertyNames) {
+        final Map<String, String> headers = new HashMap<>();
+        for (final String headerKey : dynamicPropertyNames) {
             String headerValue = context.getProperty(headerKey)
```

Comment:
Could be `final`

Reply:
But then maybe we don't want to go through and make everything `final` in all of these classes at this point?

Reply:
I'll hit what I can find in these classes, might as well update them as I go.